templates/server-service.yaml

```diff
@@ -20,7 +16,7 @@ metadata:
     # https://github.com/kubernetes/kubernetes/issues/58662
     service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
 spec:
-  clusterIP: None
+  clusterIP: {{ .Values.server.service.clusterIP }}
```

Comment:
This doesn't work if clusterIP is not set in values.yaml (and it's not by default)

I think you need an if statement before that to only include clusterIP if clusterIP is defined.

Reply:
fyi, In my case I don't want to specify a ClusterIP and I don't want it set to None.

Reply:
Interesting that you're receiving an error because it works fine on GKE when the value is empty.

Which version of Kube are you using?

Reply:
Ah, I see the error doesn't appear until you try to upgrade from a service from an earlier version.

Thanks for pointing this out, will fix this @kuuji

Reply:
Actually, after double checking, I'm only getting it on upgrade, because I guess you can't remove that field once you set it.

Reply:
I have a branch with a fix if you want @jasonodonnell . I can open a PR